sources/packages/backend/apps/workers/src/controllers/supporting-user/supporting-user.controller.ts

```diff
@@ -61,7 +69,33 @@ export class SupportingUserController {
         (supportingUser) => supportingUser.id,
       );
       jobLogger.log("Created supporting users.");
-      return job.complete({ createdSupportingUsersIds });
+      return this.dataSource.transaction(
+        async (transactionalEntityManager: EntityManager) => {
+          const application = await this.applicationService.getApplicationById(
+            job.variables.applicationId,
+          );
```

Comment:
Please move this code of getting application outside the transaction. Because if any DB operation inside the transaction is using global entity manager instead of transaction entity manager it could result deadlock connection waiting for each other as per our observation from the previous load tests.